api/context.go

```diff
@@ -105,12 +107,18 @@ func (c apicontext) WithContext(ctx gocontext.Context) Context {
 	return &c
 }
 
-func (c apicontext) WithTimeout(timeout time.Duration) (Context, func()) {
+func (c apicontext) WithTimeout(timeout time.Duration) (Context, gocontext.CancelFunc) {
 	ctx, cancel := gocontext.WithTimeout(c.Context, timeout)
 	c.Context = ctx
 	return &c, cancel
 }
 
+func (c apicontext) WithDetatchedTimeout(timeout time.Duration) (Context, gocontext.CancelFunc) {
```

Comment:
Can we remove this - I would prefer to only start using 1.21 API's when we know its working afterting deploying to demo etc..